lib/bdev/nvme/blockdev_nvme.c

```diff
@@ -502,16 +503,59 @@ probe_cb(void *cb_ctx, const struct spdk_nvme_transport_id *trid,
 }
 
 static void
-blockdev_nvme_timeout_cb(void *cb_arg, struct spdk_nvme_ctrlr *ctrlr,
-			 struct spdk_nvme_qpair *qpair, uint16_t cid)
+abort_cb(void *arg, const struct spdk_nvme_cpl *cpl)
+{
+	/*
+	 * abort completion callback function. Currently, there is nothing to do
+	 * struct spdk_nvme_ctrlr *ctrlr = arg;
+	 */
+}
+
+static void
+blockdev_nvme_timeout_cb(struct spdk_nvme_ctrlr *ctrlr,
+			 struct spdk_nvme_qpair *qpair,
+			 void *cb_arg,
```

Comment:
I understand what you're doing with the submit_ticks and it will definitely work, my questions are all coming from the perspective of trying to find the best way to handle this for everyone using the NVMe driver. I'm thinking that instead of a timeout_count in the tracker, we should put a flag in the tracker for whether a tracker is considered "timed out". The timeout callback will only be called once for a tracker's lifetime then. I'd then move the code that previously ran when timeout_count was greater than 1 to the abort completion handler. If the abort fails, reset the controller.

Let's definitely add the handling for the abort command limit too, as a separate PR.